### PR TITLE
Add signal group for dbc parsing

### PR DESCRIPTION
**Pull request by who-youknow**
Now parsing the signal group in dbc file is available.

### CODE REVIEW

**Comment by eerimoq** [comment]
Thanks, looks good!

Please add a short description about what a signal group is is the SignalGroup class docstring, as I think many users would find that information helpful.

Any chance you can add tests? A test could be to load a DBC-file with signal groups, verify them, and then dump and check that the new DBC-file matches the old old.

**Comment by who-youknow** [comment]
Hello, this is my first time to do such a job. I'll try my best to do that.
I just wonder why the AppVeyor failed. Could you give me some hint about that?

**Comment by eerimoq** [comment]
This add another line break to the dumped DBC-file: 

https://github.com/eerimoq/cantools/pull/192/files#diff-543cd2399df521fdfb65647c1848710bL84-L89

Add another blank line to the affected reference test files. They are found in ``tests/files/dbc``.

**Comment by who-youknow** [comment]
Hello I have add the description and test.
However, the unit test failed because of AssertionError.
I did not modify anything on that. Could please help me to find the reason? Thank you in advance.

**Comment by eerimoq** [comment]
I answered it in my previous comment.

**Comment by who-youknow** [comment]
Sorry, I just do not understand your hint. Could you please explain it more clearly?

**Comment by eerimoq** [comment]
Just add another `\r\n` at the end of https://github.com/eerimoq/cantools/blob/master/tests/files/dbc/attributes.dbc, and the test `test_attributes` should pass.

Do the same for other tests with the same error.

**Comment by who-youknow** [comment]
Thank you. The blank line issue is solved. However, there is still another similar issue with the degree character &#176;(°).

**Comment by eerimoq** [comment]
It's a text encoding problem most likely. You can remove that in the PR and I can re-add it after the merge.

**Comment by who-youknow** [comment]
It seems okay now except the coverage check.

**Comment by eerimoq** [comment]
Ok, I'll have a look.

File: tests/test_database.py [hunk 1]
```diff
@@ -4756,6 +4756,16 @@ def test_dbc_modify_names(self):
             else:
                 self.assertEqual(db.as_dbc_string(), fin.read())
 
+    def test_dbc_with_signal_groups(self):
+        """Test the tool can obtain the signal groups of the messages in the dbc file and dump it.
+        
+        """
+
+        filename = 'tests/files/dbc/sig_groups.dbc' #r'tests\files\dbc\sig_groups.dbc'
+        filename_dump = 'tests/files/dbc/sig_groups_gen.dbc' #r'tests\files\dbc\sig_groups_gen.dbc'
```
Comment (eerimoq) [inline tests/test_database.py:4765]: Remove comments.

File: tests/test_database.py [hunk 2]
```diff
@@ -4756,6 +4756,16 @@ def test_dbc_modify_names(self):
...
+        db = cantools.database.load_file(filename)
+        cantools.database.dump_file(db, filename_dump)
```
Comment (eerimoq) [inline tests/test_database.py:4767]: Assert that the message signal groups are as expected before dumping to file. Also, the contents of the dumped file should be checked.

File: cantools/database/can/signal.py [hunk 3]
```diff
@@ -417,3 +417,60 @@ def __repr__(self):
...
+
+        return self._signal_names
+
+    @signal_names.setter
+    def signal_names(self, value):
+        self._signal_names = value
+    
+    def __repr__(self):
+
+        return "signal group('{}', {}, {})".format(
```
Comment (eerimoq) [inline cantools/database/can/signal.py:473]: Remove the blank line.

File: cantools/database/can/formats/dbc.py [hunk 4]
```diff
@@ -1254,6 +1295,15 @@ def get_message_name(frame_id_dbc, name):
             return message_attributes['SystemMessageLongSymbol'].value
         except (KeyError, TypeError):
             return name
+    
+    def get_signal_groups(frame_id_dbc):
+        """
+        """
```
Comment (eerimoq) [inline cantools/database/can/formats/dbc.py:1301]: Remove the empty docstring.

**Pull request closed without merging**